src/app/shared/components/guided-tour/entity.ts

```diff
@@ -0,0 +1,31 @@
+// Copyright 2020 The Kubermatic Kubernetes Platform contributors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {InjectionToken, ViewContainerRef, ElementRef} from '@angular/core';
+import {DialogComponent} from './dialog/component';
+import {GuidedTourID} from './utils';
+
+export class GuidedTourStep {
+  id: string;
+  withBackdrop: boolean;
+  position: string;
+  targetViewContainer: ViewContainerRef;
+  elementRef: ElementRef;
+  stepInstance: DialogComponent;
+}
+
+export const STEP_DATA = new InjectionToken<GuidedTourStep>('STEP_DATA');
+
+export const STEP_ORDER: string[] = [
```

Comment:
This shouldn't be required. Having to define the order in code for every tour adds an additional layer of complexity for us when maintaining and creating new tours. It probably could be handled as a directive input parameter.

Second thing is that it is defined as an array of strings where actually it is an array of `GuidedTourID`.